components/server/ee/src/prebuilds/github-app.ts

```diff
@@ -423,14 +434,37 @@ export class GithubApp {
             user = r.user;
             project = r.project;
 
-            const prebuildStartPromise = await this.onPrStartPrebuild({ span }, ctx, config, context, user, project);
-            if (prebuildStartPromise) {
-                await this.onPrAddCheck({ span }, config, ctx, prebuildStartPromise);
+            await this.webhookEvents.updateEvent(event.id, {
+                authorizedUserId: user.id,
+                projectId: project?.id,
+                cloneUrl: context.repository.cloneUrl,
+                branch: context.ref,
+                commit: context.revision,
+            });
+
+            const prebuildStartResult = await this.onPrStartPrebuild({ span }, ctx, config, context, user, project);
+            if (prebuildStartResult) {
+                await this.webhookEvents.updateEvent(event.id, {
+                    prebuildStatus: "prebuild_triggered",
+                    status: "processed",
+                    prebuildId: prebuildStartResult.prebuildId,
+                });
+
+                await this.onPrAddCheck({ span }, config, ctx, prebuildStartResult);
                 this.onPrAddBadge(config, ctx);
                 await this.onPrAddComment(config, ctx);
+            } else {
+                await this.webhookEvents.updateEvent(event.id, {
+                    prebuildStatus: "ignored_unconfigured",
+                    status: "processed",
+                });
             }
         } catch (e) {
             TraceContext.setError({ span }, e);
+            await this.webhookEvents.updateEvent(event.id, {
+                prebuildStatus: "prebuild_trigger_failed",
+                status: "processed",
```

Comment:
Should the status be `processed` here? We failed to do something, do we know it will actually be created?